pkg/syslog/client.go

```diff
@@ -0,0 +1,44 @@
+package syslog
+
+import (
+	"net"
+
+	"github.com/flynn/flynn/pkg/syslog/rfc5424"
+	"github.com/flynn/flynn/pkg/syslog/rfc6587"
+)
+
+type Client struct {
+	c net.Conn
+
+	h *rfc5424.Header
+	w *rfc5424.Writer
+}
+
+func Dial(address string, hdr *rfc5424.Header) (*Client, error) {
+	conn, err := net.Dial("tcp", address)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{
+		c: conn,
+		h: hdr,
+		w: rfc5424.NewWriter(rfc6587.NewWriter(conn)),
```

Comment:
ah, you can't pre-serialize the header fully because the timestamp changes. In that case it's a little odd that you're passing in an `*rfc5424.Header` here at all since not all of its fields are actually used in the resulting message.